src/applications/disability-benefits/all-claims/components/ArrayField.jsx

```diff
@@ -0,0 +1,427 @@
+      });
+    }
+  }
+
+  /*
+   * Clicking Remove on an item in edit mode
+   */
+  handleRemove(indexToRemove) {
+    let newItems = this.props.formData.filter(
+      (val, index) => index !== indexToRemove,
+    );
+    let newState = _.assign(this.state, {
+      editing: this.state.editing.filter(
+        (val, index) => index !== indexToRemove,
+      ),
+    });
+    // Ensure that there is always a blank item in editing mode
+    if (newItems.length === 0) {
+      newItems = [
+        getDefaultFormState(
+          this.props.schema.additionalItems,
+          undefined,
+          this.props.registry.definitions,
+        ) || {},
+      ];
+      newState = _.assign(newState, {
+        editing: [true],
+      });
+    }
+    this.props.onChange(newItems);
+    this.setState(newState, () => {
+      this.scrollToTop();
+    });
+  }
+
+  render() {
+    const {
+      uiSchema,
+      errorSchema,
+      idSchema,
+      formData,
+      disabled,
+      readonly,
+      registry,
+      formContext,
+      onBlur,
+      schema,
+    } = this.props;
+    const definitions = registry.definitions;
+    const { TitleField, SchemaField } = registry.fields;
+
+    const uiOptions = uiSchema['ui:options'] || {};
+    const ViewField = uiOptions.viewField;
+    const title = uiSchema['ui:title'] || schema.title;
+    const hideTitle = !!uiOptions.title;
+    const description = uiSchema['ui:description'];
+    const textDescription =
+      typeof description === 'string' ? description : null;
+    const DescriptionField =
+      typeof description === 'function' ? uiSchema['ui:description'] : null;
+    const hasTitleOrDescription = (!!title && !hideTitle) || !!description;
+
+    // if we have form data, use that, otherwise use an array with a single default object
+    const items =
+      formData && formData.length
+        ? formData
+        : [getDefaultFormState(schema, undefined, registry.definitions)];
+
+    const containerClassNames = classNames({
+      'schemaform-field-container': true,
+      'schemaform-block': hasTitleOrDescription,
+    });
+
+    return (
+      <div className={containerClassNames}>
+        {hasTitleOrDescription && (
+          <div className="schemaform-block-header">
+            {title && !hideTitle ? (
+              <TitleField
+                id={`${idSchema.$id}__title`}
+                title={title}
+                formContext={formContext}
+              />
+            ) : null}
+            {textDescription && <p>{textDescription}</p>}
+            {DescriptionField && (
+              <DescriptionField options={uiSchema['ui:options']} />
+            )}
+            {!textDescription && !DescriptionField && description}
+          </div>
+        )}
+
+        <div className="va-growable">
+          <Element name={`topOfTable_${idSchema.$id}`} />
+          {items.map((item, index) => {
+            // This is largely copied from the default ArrayField
```

Comment:
This comment comes from the USFS `ArrayField`, which is a bit misleading now, since it kinda sounds like the "default ArrayField" = USFS `ArrayField`, but it really means RJSF's `ArrayField`.